pkg/controller/provider/controller.go

```diff
@@ -334,6 +354,155 @@ func (r *Reconciler) getSecret(provider *api.Provider) (*core.Secret, error) {
 	return secret, nil
 }
 
+func (r *Reconciler) createOVAServerDeployment(provider *api.Provider, ctx context.Context) {
+	deploymentName := fmt.Sprintf("%s-deployment-%s", ovaServerPrefix, provider.UID)
+
+	deployment := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), client.ObjectKey{
+		Namespace: provider.Namespace,
+		Name:      deploymentName},
+		deployment)
+
+	//if the deployent not exist
+	if k8serr.IsNotFound(err) {
+		annotations := make(map[string]string)
+		labels := map[string]string{"providerName": provider.Name, "app": "forklift"}
+		url := provider.Spec.URL
+		var replicas int32 = 1
+
+		//OVA server deployment
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        deploymentName,
+				Namespace:   provider.Namespace,
+				Annotations: annotations,
+				Labels:      labels,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "forklift",
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"providerName": provider.Name,
+							"app":          "forklift",
+						},
+					},
+					Spec: makeOvaProviderPodSpec(url, string(provider.UID)),
+				},
+			},
+		}
+
+		err = r.Create(ctx, deployment)
+		if err != nil {
+			return
+		}
+
+		// OVA Server Service
+		serviceName := fmt.Sprintf("ova-service-%s", provider.UID)
+		service := &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceName,
+				Namespace: provider.Namespace,
+				Labels:    labels,
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{
+					"providerName": provider.Name,
+					"app":          "forklift",
+				},
+				Ports: []v1.ServicePort{
+					{
+						Name:       "api-http",
+						Protocol:   v1.ProtocolTCP,
+						Port:       8080,
+						TargetPort: intstr.FromInt(8080),
+					},
+				},
+				Type: v1.ServiceTypeClusterIP,
+			},
+		}
+
+		err = r.Create(ctx, service)
+		if err != nil {
+			return
+		}
+
+		secret, err1 := r.getSecret(provider)
+		if err1 != nil {
+			return
+		}
+
+		podURL := fmt.Sprintf("http://%s:8080", serviceName)
+		secret.Data["pod-url"] = []byte(podURL)
+		err = r.Update(ctx, secret)
```

Comment:
since the only place using this value is in the Ova client, I added a new field called serviceURL in it and when it bein initialize (and test connection) I'm saving this value for all future use, using it as part of the status is a bit more complicated since its being override.

Reply:
ok, we'll look at it later then